File: sdk/java-sdk/src/test/java/com/evernym/verity/sdk/protocols/IssuerSetupTest.java
package com.evernym.verity.sdk.protocols;

import com.evernym.verity.sdk.TestHelpers;
import com.evernym.verity.sdk.exceptions.VerityException;
import com.evernym.verity.sdk.protocols.issuersetup.IssuerSetup;
import com.evernym.verity.sdk.protocols.issuersetup.v0_7.IssuerSetupV0_7;
import com.evernym.verity.sdk.utils.Context;
import com.evernym.verity.sdk.utils.Util;
import org.json.JSONObject;
import org.junit.jupiter.api.Test;

import static com.evernym.verity.sdk.TestHelpers.unpackForwardMessage;
import static org.junit.jupiter.api.Assertions.*;

public class IssuerSetupTest {

    @Test
    public void testGetMessageType() {
        String msgName = "msg name";

        IssuerSetupV0_7 t = IssuerSetup.v0_7();

        String expectedType = Util.getMessageType(
                Util.EVERNYM_MSG_QUALIFIER,
                t.family(),
                t.version(),
                msgName);

        assertEquals(expectedType, t.messageType(msgName));
    }

    @Test
    public void testGetThreadId() {
        IssuerSetupV0_7 testProtocol = IssuerSetup.v0_7();
        assertNotNull(testProtocol.getThreadId());
    }

    @Test
    public void testCreateMessages() throws VerityException {
        Context context = TestHelpers.getContext();
        IssuerSetupV0_7 p = IssuerSetup.v0_7();
        JSONObject msg = p.createMsg(context, "did:indy:sovrin:builder");
        assertEquals(
                Util.EVERNYM_MSG_QUALIFIER + "/issuer-setup/0.7/create",
                msg.getString("@type")
        );
        asserEquals(
                msg.getString('ledgerPrefix'),
                "did:indy:sovrin:builder"
        );
        assertNotNull(msg.getString("@id"));
    }

    @Test
    public void testCreateMessages() throws VerityException {
        Context context = TestHelpers.getContext();
        IssuerSetupV0_7 p = IssuerSetup.v0_7();
        JSONObject msg = p.createMsg(context, "did:indy:sovrin:builder", "someEndorser");
        assertEquals(
                Util.EVERNYM_MSG_QUALIFIER + "/issuer-setup/0.7/create",
                msg.getString("@type")
        );
        asserEquals(
                msg.getString('ledgerPrefix'),
                "did:indy:sovrin:builder"
        );
        asserEquals(
                msg.getString(endorser),
                "someEndorser"
        );
        assertNotNull(msg.getString("@id"));
    }

    @Test
    public void testCreate() throws Exception {
        Context context = null;
        try {
            context = TestHelpers.getContext();
            IssuerSetupV0_7 testProtocol = IssuerSetup.v0_7();
            byte[] message = testProtocol.createMsgPacked(context, "did:indy:sovrin:builder");
            JSONObject unpackedMessage = unpackForwardMessage(context, message);
            assertEquals(
                    Util.EVERNYM_MSG_QUALIFIER + "/issuer-setup/0.7/create",
                    unpackedMessage.getString("@type")
            );

        } catch(Exception e) {
            e.printStackTrace();
            fail();
        } finally {
            TestHelpers.cleanup(context);
        }
    }
}

File: sdk/dotnet-sdk/src/VeritySDK.Test/Protocols/IssuerSetup/IssuerSetupTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Json;
using VeritySDK.Protocols.IssuerSetup;
using VeritySDK.Utils;

namespace VeritySDK.Test
{
    [TestClass]
    public class IssuerSetupTest : TestBase
    {

        [TestMethod]
        public void testGetMessageType()
        {
            string msgName = "msg name";

            IssuerSetupV0_7 t = IssuerSetup.v0_7();

            string expectedType = Util.getMessageType(
                    Util.EVERNYM_MSG_QUALIFIER,
                    t.family(),
                    t.version(),
                    msgName);

            Assert.AreEqual(expectedType, t.messageType(msgName));
        }

        [TestMethod]
        public void testGetThreadId()
        {
            IssuerSetupV0_7 testProtocol = IssuerSetup.v0_7();
            Assert.IsNotNull(testProtocol.getThreadId());
        }

        [TestMethod]
        public void testCreateMessages()
        {
            Context context = TestHelpers.getContext();
            IssuerSetupV0_7 p = IssuerSetup.v0_7();
            JsonObject msg = p.createMsg(context, "did:indy:sovrin:builder");
            Assert.AreEqual(
                    Util.EVERNYM_MSG_QUALIFIER + "/issuer-setup/0.7/create",
                    msg.getAsString("@type")
            );
            Assert.AreEqual(
                msg.getAsString("ledgerPrefix"),
                "did:indy:sovrin:builder"
            );
            Assert.IsNotNull(msg.getAsString("@id"));
        }

        public void testCreateMessagesWithEndorser()
        {
            Context context = TestHelpers.getContext();
            IssuerSetupV0_7 p = IssuerSetup.v0_7();
            JsonObject msg = p.createMsg(context, "did:indy:sovrin:builder", "someEndorser");
            Assert.AreEqual(
                    Util.EVERNYM_MSG_QUALIFIER + "/issuer-setup/0.7/create",
                    msg.getAsString("@type")
            );
            Assert.AreEqual(
                msg.getAsString("ledgerPrefix"),
                "did:indy:sovrin:builder"
            );
            Assert.AreEqual(
                 msg.getAsString("endorser"),
                 "someEndorser"
             );
            Assert.IsNotNull(msg.getAsString("@id"));
        }

        [TestMethod]
        public void testCreate()
        {
            withContext(context =>
            {
                IssuerSetupV0_7 testProtocol = IssuerSetup.v0_7();
                byte[] message = testProtocol.createMsgPacked(context, "did:indy:sovrin:builder");
                JsonObject unpackedMessage = TestHelpers.unpackForwardMessage(context, message);
                Assert.AreEqual(
                        Util.EVERNYM_MSG_QUALIFIER + "/issuer-setup/0.7/create",
                        unpackedMessage.getAsString("@type"));
            });
        }
    }
}
